lib/game.rb

```diff
@@ -0,0 +1,97 @@
+class Game
+  attr_accessor :player_1, :player_2, :board
+  extend Players
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def input_to_index(input)
+    input.to_i - 1
+  end
+
+  def current_player
+    @board.turn_count % 2 == 0 ? @player_1 : @player_2
+  end
+
+  WIN_COMBINATIONS = [
+    [0,1,2],
+    [3,4,5],
+    [6,7,8],
+    [0,3,6],
+    [1,4,7],
+    [2,5,8],
+    [0,4,8],
+    [6,4,2]
+  ]
+
+  def turn
+    loop do
+      input = input_to_index(current_player.move(@board))
+      if @board.valid_move?(input + 1) == true
+        @board.cells[input] = current_player.token
+        @board.display
+        break
+      end
+    end
+  end
+
+  def won?
+      WIN_COMBINATIONS.each do |win_combo|
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.detect do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]] &&
      @board.taken?(combo[0] + 1)
    end
  end
```